app/workers/parser_site_worker.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+class ParserSiteWorker
+  include Sidekiq::Worker
+
+  def perform
+    search_patterns = Serverversion.pluck(:name)
+    sites = ParserSite.where(enabled: true)
+    sites.each do |site|
+      parsing_site(site, search_patterns)
+    end
+  end
+
+  def parsing_site(site, search_patterns)
+    document = Services::ParserEngine::GettingDataSite.new(site.url).take
+    return if document.blank?
+
+    selected_data = Services::ParserEngine::DataPreparation.new(document, site.css_selector).selected
+    array_of_match_numbers = Services::ParserEngine::SearchByPatterns.new(selected_data, search_patterns).collect_array
+    array_of_new_servers = Services::ParserEngine::MakingServersFromData.new(array_of_match_numbers, selected_data, site).make
```

Comment:
this can be ServersBuilder

Reply:
Also I think if we can add one more service that will combine all these functions... and is it needed